remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java

```diff
@@ -95,17 +99,31 @@ public void onClick(View view) {
    */
 
   public void showRemixer(FragmentManager manager, String tag) {
-    if (!isAdded() && !isVisible() && !isRemoving()) {
-      show(manager, tag);
+    synchronized(syncLock) {
+      if (!isAddingFragment && !isAdded()) {
+        isAddingFragment = true;
+        show(manager, tag);
+      }
     }
   }
-  
+
+  // TODO(nicksahler): Generalize to attaching to any SensorEventListener
+
   /**
    * Attach this instance to a shake gesture and show fragment when magnitude exceeds {@code threshold}
    *
    */
   public void attachToShake(final FragmentActivity activity, final double threshold) {
-    ShakeListener.attach(activity, threshold, this);
+    shakeListener = new ShakeListener(activity, threshold, this);
+    shakeListener.attach();
+  }
+
+  /**
+   * Detach from a shake gesture
+   *
+   */
+  public void detachFromShake(final FragmentActivity activity) {
+    shakeListener.detach();
```

Comment:
Let's be extra careful with leaking stuff, though right now this doesn't leak anything (that I can see) add a shakeListener = null; after this.